azurelinuxagent/common/event.py

```diff
@@ -282,6 +281,51 @@ def _log_event(name, op, message, duration, is_success=True): # pylint: disable=
         logger.info(_EVENT_MSG, name, op, message, duration)
 
 
+class EventDebugInfo(object):
+    MAX_ERRORS = 5
+    OP_REPORT = "Report"
+    OP_COLLECT = "Collect"
+
+    def __init__(self, operation=OP_REPORT):
+        self.unicode_error_count = 0
+        self.unicode_errors = set()
+        self.op_error_count = 0
+        self.op_errors = set()
+
+        if operation == self.OP_REPORT:
+            self.unicode_error_event = WALAEventOperation.ReportEventUnicodeErrors
+            self.op_errors_event = WALAEventOperation.ReportEventErrors
+        elif operation == self.OP_COLLECT:
+            self.unicode_error_event = WALAEventOperation.CollectEventUnicodeErrors
+            self.op_errors_event = WALAEventOperation.CollectEventErrors
+
+    def report_debug_info(self):
+
+        def report_dropped_events_error(count, errors, operation_name):
+            err_msg_format = "DroppedEventsCount: {0}\nReasons (first {1} errors): {2}"
+            if count > 0:
+                add_event(op=operation_name,
+                          message=err_msg_format.format(count, EventDebugInfo.MAX_ERRORS, ', '.join(errors)),
+                          is_success=False)
+
+        report_dropped_events_error(self.op_error_count, self.op_errors, self.op_errors_event)
+        report_dropped_events_error(self.unicode_error_count, self.unicode_errors, self.unicode_error_event)
+
+    @staticmethod
+    def _update_errors_and_get_count(error_count, errors, error):
+        error_count += 1
```

Comment:
what is the purpose of 'error_count'? should it be incremented only of the error was added? although the return value of the function doesn't seem to be used anyways... :)

Reply:
Or did you mean to use self.op_error_count? that one is initialized to 0 and never changes